src/Database/LSMTree/Internal/IndexCompact.hs

```diff
@@ -436,42 +436,67 @@ pageSpanSize :: PageSpan -> NumPages
 pageSpanSize pspan = NumPages . toEnum $
     unPageNo (pageSpanEnd pspan) - unPageNo (pageSpanStart pspan) + 1
 
--- | Given a search key, find the number of the disk page that /might/ contain
--- this key.
+-- |  Searches for a page span that contains a key–value pair with the given key.
 --
--- Note: there is no guarantee that the disk page contains the key. However, it
--- is guaranteed that /no other/ disk page contain the key.
+-- If there is indeed such a pair, the result is the corresponding page span; if
+-- there is no such pair, the result is an arbitrary but valid page span.
 --
 -- See [an informal description of the search algorithm](#search-descr) for more
 -- details about the search algorithm.
---
--- TODO: optimisation ideas: we should be able to get it all unboxed,
--- non-allocating and without unnecessary bounds checking.
```

Comment:
INFO: It's actually already non-allocating! There are probably still other optimisation opportunities, but I'm removing the comment here